proto/api/v1/remote_runner.proto

```diff
@@ -28,6 +28,17 @@ message RunRequest {
   // Ex. {"OSFamily":"linux", "Arch":"amd64"}
   map<string, string> platform_properties = 6;
 
+  // Platform properties that should be set via remote header override.
+  // These apply to the remote runner (and not the commands run by the runner).
+  //
+  // These are similar to `platform_properties`, except they will not be
+  // included in the snapshot key (which contains regular platform properties)
+  // and won't be displayed on the executions page.
+  //
+  // They are useful for passing secrets that you don't want displayed or
+  // short-lived credentials that you don't want invalidating the snapshot key.
+  map<string, string> remote_header_overrides = 10;
```

Comment:
I think I would prefer calling this `remote_headers` and blindly pass them through as gRPC headers in the Execute request. So people would have to set `"x-buildbuddy-platform.NAME": "value"` rather than just `"name": "value"`. This would allow people to set whatever headers they want, not just platform overrides, and is also more consistent with our docs on how to pass secret headers today (`--remote_header=x-buildbuddy-platform.SECRET_NAME=SECRET_VALUE`)

If you were aiming for something more user friendly, then another option is to call this `secret_platform_properties` and avoid mentioning anything about remote headers, since `remote_header_overrides` seems like a bit of an implementation detail, rather than directly communicating the intended purpose. Personally though, I would go with `remote_headers` since it's more flexible and is also more consistent with the normal RBE APIs.

Reply:
Good call - updated